crates/kornia-apriltag/src/iter.rs

```diff
@@ -0,0 +1,285 @@
+use crate::{
+    errors::AprilTagError,
+    utils::{find_total_tiles, Point2d},
+};
+use kornia_image::{allocator::ImageAllocator, Image, ImageSize};
+
+/// Contains metadata and data for a single tile of an image.
+///
+/// `TileInfo` holds the position, indices, and a borrowed slice of pixel data for a tile.
+/// The `data` field is a borrowed slice of rows, where each row is a slice of pixel data.
+#[derive(Debug, Clone, PartialEq)]
+pub struct TileInfo<'a, T> {
+    /// The 2D position (x, y) of the tile in tile coordinates.
+    pub pos: Point2d,
+    /// The sequential index of the tile (including partial tiles).
+    pub index: usize,
+    /// The index among full (non-partial) tiles.
+    pub full_index: usize,
+    /// A borrowed slice of rows, where each row is a slice of pixel data.
+    pub data: &'a [&'a [T]],
+}
+
+#[derive(Debug, Clone, PartialEq)]
+/// Represents a tile of an image, which can be either a full-sized tile or a partial tile at the image edge.
+///
+/// Each variant contains a slice of rows, where each row is a slice of pixel data.
+pub enum ImageTile<'a, T> {
+    /// A full-sized tile with dimensions equal to the specified tile size.
+    FullTile(TileInfo<'a, T>),
+    /// A partial tile, typically at the image edge, with dimensions smaller than the specified tile size.
+    PartialTile(TileInfo<'a, T>),
+}
+
+/// An iterator over tiles of an image, yielding non-overlapping rectangular regions as slices.
+///
+/// Each item yielded by the iterator is a slice of rows, where each row is a slice of pixel data.
+/// The tile size is specified at construction, and the iterator will yield tiles of the given size,
+/// except for tiles at the image edges, which may be smaller if the image dimensions are not
+/// multiples of the tile size.
+pub struct TileIterator<'a, T> {
+    img_data: &'a [T],
+    img_size: ImageSize,
+    tile_size: usize,
+    tiles_len: Point2d,
```

Comment:
`tiles_len` suggests a scalar while this is a `Point2d`. For readability I suggest renaming it to something like `tiles_dim`